File: core/src/ru/myitschool/dcrawler/entities/ai/task/AnalyzeTask.java
package ru.myitschool.dcrawler.entities.ai.task;

import ru.myitschool.dcrawler.entities.ai.AI;

public abstract class AnalyzeTask extends Task {

    private AI ai;

    public AnalyzeTask(AI ai) {
        this.ai = ai;
    }

    @Override
    protected void startTask() {
        analyze();
        complete();
    }

    protected abstract void analyze();
}

File: core/src/com/wolg_vlad/dcrawler/entities/ai/task/AnalyzeTask.kt
package com.wolg_vlad.dcrawler.entities.ai.task

import com.wolg_vlad.dcrawler.entities.ai.AI

abstract class AnalyzeTask(private val ai: AI) : Task() {
    override fun startTask() {
        analyze()
        complete()
    }

    protected abstract fun analyze()

}